frontend/src/components/SharedGoalComponents/ObjectiveSelection.js

```diff
@@ -0,0 +1,152 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Controller, useFormContext } from 'react-hook-form';
+import Select from 'react-select';
+import { Button, Textarea } from '@trussworks/react-uswds';
+import FormItem from '../FormItem';
+import selectOptionsReset from '../selectOptionsReset';
+import { CREATE_A_NEW_OBJECTIVE } from './constants';
+
+const ObjectiveTextArea = ({ fieldName, index, fieldValue }) => {
+  const { register } = useFormContext();
+  return (
+    <Textarea
+      name={`${fieldName}[${index}].value`}
+      id={`${fieldName}[${index}].value`}
+      className="margin-bottom-1"
+      inputRef={register()}
+      defaultValue={fieldValue}
+      required
+    />
+  );
+};
+
+ObjectiveTextArea.propTypes = {
+  fieldName: PropTypes.string.isRequired,
+  index: PropTypes.number.isRequired,
+  fieldValue: PropTypes.string,
+};
+
+ObjectiveTextArea.defaultProps = {
+  fieldValue: '',
+};
+
+export default function ObjectiveSelection({
+  field,
+  index,
+  remove,
+  fieldName,
+  objectiveOptions,
+}) {
+  const { setValue, register, watch } = useFormContext();
+
+  const selectedObjectives = watch(fieldName);
+  const selectedObjectiveTitles = useMemo(() => (
+    selectedObjectives.map((o) => o.label)
+  ), [selectedObjectives]);
+  const fieldLabel = watch(`${fieldName}[${index}].label`);
+
+  const filteredOptions = objectiveOptions.filter((option) => {
+    if (option.label === fieldLabel) {
+      return true;
+    }
+
+    return !(selectedObjectiveTitles.includes(option.label));
+  });
+
+  const onlyCreateNew = (
+    filteredOptions.length === 1
+    && filteredOptions[0].label === CREATE_A_NEW_OBJECTIVE
+  );
+
+  return (
+    <div key={field.id}>
+      <input
+        type="hidden"
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register(`${fieldName}[${index}].objectiveId`)}
+        defaultValue={field.objectiveId}
+      />
+      <input
+        type="hidden"
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register(`${fieldName}[${index}].value`)}
+        defaultValue={field.objectiveId}
+      />
```

Comment:
Both hidden input `defaultValue`s are set to `field.objectiveId`. Is that intentional?

Reply:
good catch thanks